extension/src/popup/components/signTransaction/Operations.tsx

```diff
@@ -0,0 +1,260 @@
+import { TransactionList } from "popup/basics/TransactionList";
+
+interface TransactionInfoResponse {
+  amount: string;
+  destination: string;
+  asset: { code: string };
+  signer: {
+    ed25519PublicKey?: string;
+    sha256Hash?: { data: Buffer };
+    preAuthTx?: { data: Buffer };
+    weight: number;
+  };
+  type: keyof typeof OPERATION_TYPES;
+  buying: { code: string };
+  selling: { code: string };
+  buyAmount: string;
+  price: string;
+  inflationDest: string;
+  setFlags: number;
+  clearFlags: number;
+  masterWeight: number;
+  lowThreshold: number;
+  medThreshold: number;
+  highThreshold: number;
+}
+
+const OperationBoxEl = styled.div`
+  text-align: left;
+`;
+const OperationBoxHeaderEl = styled.h4`
+  color: ${COLOR_PALETTE.primary};
+  font-size: 1.25rem;
+  font-weight: ${FONT_WEIGHT.normal};
+  margin: 0;
+  padding: 0;
+  padding-top: 1.875rem;
+  padding-bottom: 0.625rem;
+  padding-left: 0.43rem;
+
+  strong {
+    font-weight: ${FONT_WEIGHT.bold};
+  }
+`;
+const OperationsListEl = styled(TransactionList)`
+  padding-left: 1.25rem;
+
+  li {
+    div {
+      width: 50%;
+
+      &:first-child {
+        padding: 0;
+        color: ${COLOR_PALETTE.text};
+      }
+    }
+  }
+`;
+
+const KeyValueList = ({
+  operationKey,
+  operationValue,
+}: {
+  operationKey: string;
+  operationValue: string | number;
+}) => (
+  <li>
+    <div>{operationKey}:</div>
+    <div>{operationValue}</div>
+  </li>
+);
+
+enum AuthorizationMap {
+  "",
+  "Authorization Required",
+  "Authorization Revocable",
+  "Authorization Required; Authorization Required",
+  "Authorization Immutable",
+  "Authorization Required; Authorization Immutable",
+  "Authorization Revocable; Authorization Immutable",
+  "Authorization Required; Authorization Required; Authorization Immutable",
+}
+
+const formattedBuffer = (data: Buffer) =>
+  truncatedPublicKey(Buffer.from(data).toString("hex").toUpperCase());
+
+export const Operations = ({
+  operations,
+}: {
+  operations: [TransactionInfoResponse];
+}) => (
+  <>
+    {operations.map(
+      (
+        {
+          amount,
+          destination,
+          asset,
+          signer,
+          type,
+          buying,
+          selling,
+          buyAmount,
+          price,
+          inflationDest,
+          setFlags,
+          clearFlags,
+          masterWeight,
+          lowThreshold,
+          medThreshold,
+          highThreshold,
+        }: TransactionInfoResponse,
+        i: number,
+      ) => {
+        const operationIndex = i + 1;
+
+        return (
+          <OperationBoxEl key={operationIndex}>
+            <OperationBoxHeaderEl>
+              {operationIndex}. {OPERATION_TYPES[type]}
+            </OperationBoxHeaderEl>
+            <OperationsListEl>
+              {destination ? (
+                <KeyValueList
+                  operationKey="Destination"
+                  operationValue={truncatedPublicKey(destination)}
+                />
+              ) : null}
```

Comment:
this pattern of checking a value and then showing a custom label/value pair is a little duplicative, but each operation param is _just_ different enough that this appears to be the cleanest, most readable way to show these